### PR TITLE
Test for sync up without local changes

### PR DESCRIPTION
**Pull request by wmathurin**
(And tweak to the code to not fire more updates than previously when they are indeed local changes)

### CODE REVIEW

File: libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m [hunk 1]
```diff
@@ -375,8 +375,8 @@ - (void) syncUp:(SFSyncState*)sync updateSync:(SyncUpdateBlock)updateSync failSy
     // Call smartstore
     NSArray* dirtyRecordIds = [target getIdsOfRecordsToSyncUp:self soupName:soupName];
     NSUInteger totalSize = [dirtyRecordIds count];
-    updateSync(nil, totalSize == 0 ? 100 : 0, totalSize, kSyncManagerUnchanged);
```
Comment (wmathurin) [inline libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m]: We introduced that line a couple of days ago - which meant one extra update than previously when there are changes.

**Comment by bhariharan** [comment]
Looks good. ```master``` or ```unstable```?

**Comment by wmathurin** [comment]
Master because I also changed the code -- i will also update the smartsync.a in the plugin repo.

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**